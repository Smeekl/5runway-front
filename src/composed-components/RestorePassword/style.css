.restore-password {
  padding: 62px 35px 0 35px;
}

.restore-password__title {
  font-weight: 700;
  font-size: 24px;
  color: var(--color-primary);
}

.restore-password__description {
  font-weight: 300;
  margin-bottom: 20px;
}

.restore-password__field {
  width: 343px;
  height: 40px;
  border: 1px solid var(--color-secondary-mad-blue);
  border-radius: 15px;
  margin-bottom: 8px;
  padding: 4px 11px;
}

.restore-password__block {
  margin-bottom: 24px;
}

.restore-password__btn {
  background: var(--color-primary-blue);
  border-radius: 15px;
  width: 100%;
  height: 48px;
  border: none;
}

.restore-password__btn > span {
  color: var(--color-secondary-white);
}

.restore-password__btn:hover,
.restore-password__btn:focus {
  background: var(--color-primary);
}
