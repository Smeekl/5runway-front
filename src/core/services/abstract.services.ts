import { AxiosInstance } from 'axios';
import { StorageService } from './storage-services/storage.services';
import { TTokens } from './types';

export class AbstractApiClient {
  api: AxiosInstance;
  storageService: StorageService;

  constructor(api: AxiosInstance, storageService: StorageService) {
    this.api = api;
    this.storageService = storageService;
  }

  setTokens({ accessToken, refreshToken }: TTokens) {
    this.storageService.set('accessToken', accessToken);
    this.storageService.set('refreshToken', refreshToken);
  }
}
