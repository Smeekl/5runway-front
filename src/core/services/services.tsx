import React from 'react';
import { IServices } from '@serviceTypes';

const { createContext, useContext } = React;
const ServicesContext = createContext({} as IServices);

type TServiceProviderProps = {
  services: IServices;
  children: React.ReactNode;
};

export const ServiceProvider = ({
  services,
  children,
}: TServiceProviderProps) => (
  <ServicesContext.Provider value={services}>
    {children}
  </ServicesContext.Provider>
);

export const useService = () => useContext(ServicesContext);
