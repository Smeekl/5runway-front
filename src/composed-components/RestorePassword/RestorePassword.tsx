import { Field, Form, Formik } from 'formik';
import { useNavigate } from 'react-router-dom';
import Button from '@components/Button/Button';
import SidebarErrorIndicator from '@components/SidebarErrorIndicator/SidebarErrorIndicator';
import { RestorePasswordSchema } from '@schemas';
import { useRestorePassword } from './useRestorePassword.hook';
import './style.css';

const RestorePassword = () => {
  const navigate = useNavigate();
  const { mutate } = useRestorePassword(() => navigate('/login/message-sent'));

  return (
    <div className="restore-password">
      <h3 className="restore-password__title">Restore Password</h3>
      <p className="restore-password__description">
        Fill in the field and a link to restore your password will be sent to
        your email.
      </p>
      <Formik
        initialValues={{ email: '' }}
        validationSchema={RestorePasswordSchema}
        onSubmit={({ email }) => {
          try {
            mutate(email);
          } catch (error) {}
        }}
      >
        {({ errors, touched, isValid, dirty }) => (
          <Form className="restore-password__form">
            <div className="restore-password__block">
              <p className="restore-password__name">Email</p>
              <Field
                name="email"
                type="email"
                className="restore-password__field"
              ></Field>
              <SidebarErrorIndicator
                content={errors.email || ''}
                active={Boolean(touched.email && errors.email)}
              />
            </div>

            <Button
              htmlType="submit"
              disabled={!(isValid && dirty)}
              className="restore-password__btn"
            >
              Restore
            </Button>
          </Form>
        )}
      </Formik>
    </div>
  );
};

export default RestorePassword;
