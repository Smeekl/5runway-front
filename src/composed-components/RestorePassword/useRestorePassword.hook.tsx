import { useMutation } from 'react-query';
import { useService } from '@services/services';

export const useRestorePassword = (cb: () => void) => {
  const { userService } = useService();

  const { data, isLoading, isError, mutate, error } = useMutation(
    'restorePassword',
    (email: string) => userService.restorePassword(email),
    {
      onSuccess: () => {
        cb();
      },
    }
  );

  return {
    data,
    error,
    isLoading,
    isError,
    mutate,
  };
};
