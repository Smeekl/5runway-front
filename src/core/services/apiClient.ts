import Axios, { AxiosInstance } from 'axios';
import { StorageService } from './storage-services/storage.services';

export function CreateApiClient(
  baseURL: string,
  storageService: StorageService
): AxiosInstance {
  const api = Axios.create({ baseURL });

  let retry = false;

  api.interceptors.request.use(
    (config) => {
      if (!config.headers) {
        config.headers = {};
      }

      config.headers.Authorization = `Bearer ${storageService.get(
        'accessToken'
      )}`;

      return config;
    },
    (error) => Promise.reject(error)
  );
  api.interceptors.response.use(
    (res) => res,
    async (err) => {
      if (err.response.status === 401 && !retry) {
        retry = true;

        const refreshToken = storageService.get('refreshToken');

        if (!refreshToken) {
          return;
        }

        try {
          const { data } = await api.post('/auth/refresh', { refreshToken });

          storageService.set('accessToken', data.accessToken);
          storageService.set('refreshToken', data.refreshToken);
        } catch (error) {
          throw new Error('Token expired');
        }
        return;
      }

      retry = false;

      return Promise.reject(err);
    }
  );

  return api;
}
