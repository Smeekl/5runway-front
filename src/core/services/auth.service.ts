import { AbstractApiClient } from './abstract.services';
import { TTokens } from './types';

export class AuthService extends AbstractApiClient {
  googleAuth = async (google_token: string) => {
    const { data } = await this.api.get<TTokens>(`/auth/google/verification?token=${google_token}`);

    this.setTokens(data);
  }
}
