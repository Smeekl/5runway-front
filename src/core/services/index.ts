import { CreateApiClient } from './apiClient';
import { StorageService } from './storage-services/storage.services';
import { UserService } from './user.service';
import { SearchService } from './search.service';
import { AuthService } from './auth.service';

export function createServices() {
  const storageService = new StorageService(window.localStorage);
  const api = CreateApiClient(process.env.REACT_APP_BASEURL, storageService);
  const searchService = new SearchService(api, storageService);
  const userService = new UserService(api, storageService);
  const authService = new AuthService(api, storageService);

  return {
    storageService,
    searchService,
    userService,
    authService,
  };
}
