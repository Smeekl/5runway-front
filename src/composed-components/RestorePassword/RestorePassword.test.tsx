import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { ServiceProvider } from '@services/services';
import { createServices } from '@services/index';
import RestorePassword from './RestorePassword';

describe('RestorePassword UI tests', () => {
  test('should render correctly', () => {
    const services = createServices();

    const { container } = render(
      <MemoryRouter>
        <ServiceProvider services={services}>
          <RestorePassword />
        </ServiceProvider>
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('should be in the document', () => {
    const services = createServices();

    const { container } = render(
      <MemoryRouter>
        <ServiceProvider services={services}>
          <RestorePassword />
        </ServiceProvider>
      </MemoryRouter>
    );
    expect(container).toBeInTheDocument();
  });
});
