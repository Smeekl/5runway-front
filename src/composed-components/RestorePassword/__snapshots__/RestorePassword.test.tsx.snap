// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestorePassword UI tests should render correctly 1`] = `
<div>
  <div
    class="restore-password"
  >
    <h3
      class="restore-password__title"
    >
      Restore Password
    </h3>
    <p
      class="restore-password__description"
    >
      Fill in the field and a link to restore your password will be sent to your email.
    </p>
    <form
      action="#"
      class="restore-password__form"
    >
      <div
        class="restore-password__block"
      >
        <p
          class="restore-password__name"
        >
          Email
        </p>
        <input
          class="restore-password__field"
          name="email"
          type="email"
          value=""
        />
        <div
          class="sidebar__error-indicator "
        >
          <span
            aria-label="exclamation-circle"
            class="anticon anticon-exclamation-circle sidebar__error-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="exclamation-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
              <path
                d="M464 688a48 48 0 1096 0 48 48 0 10-96 0zm24-112h48c4.4 0 8-3.6 8-8V296c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8z"
              />
            </svg>
          </span>
        </div>
      </div>
      <button
        class="ant-btn ant-btn-default restore-password__btn"
        disabled=""
        type="submit"
      >
        <span>
          Restore
        </span>
      </button>
    </form>
  </div>
</div>
`;

exports[`RestorePasswordSidebar UI tests should render correctly 1`] = `
<div>
  <div
    class="restore-password"
  >
    <h3
      class="restore-password__title"
    >
      Restore Password
    </h3>
    <p
      class="restore-password__description"
    >
      Fill in the field and a link to restore your password will be sent to your email.
    </p>
    <form
      action="#"
      class="restore-password__form"
    >
      <div
        class="restore-password__block"
      >
        <p
          class="restore-password__name"
        >
          Email
        </p>
        <input
          class="restore-password__field"
          name="email"
          type="email"
          value=""
        />
        <div
          class="sidebar__error-indicator "
        >
          <span
            aria-label="exclamation-circle"
            class="anticon anticon-exclamation-circle sidebar__error-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="exclamation-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
              <path
                d="M464 688a48 48 0 1096 0 48 48 0 10-96 0zm24-112h48c4.4 0 8-3.6 8-8V296c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8z"
              />
            </svg>
          </span>
        </div>
      </div>
      <button
        class="ant-btn ant-btn-default restore-password__btn"
        disabled=""
        type="submit"
      >
        <span>
          Restore
        </span>
      </button>
    </form>
  </div>
</div>
`;
